### PR TITLE
KER-291: Login Component

### PR DESCRIPTION
**Pull request by arpargo**
Updated new Login component so users authenticate using Helsinki Profiili instead of Tunnistamo.

### CODE REVIEW

**Comment by terovirtanen** [comment]
**KERROKANTASI-UI branch is deployed to platta: https://kerrokantasi-pr996-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://kerrokantasi-pr996-ui.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KERROKANTASI-UI branch is deployed to platta: https://kerrokantasi-pr996-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://kerrokantasi-pr996-ui.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KERROKANTASI-UI branch is deployed to platta: https://kerrokantasi-pr996-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://kerrokantasi-pr996-ui.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KERROKANTASI-UI branch is deployed to platta: https://kerrokantasi-pr996-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**KERROKANTASI-UI branch is deployed to platta: https://kerrokantasi-pr996-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://kerrokantasi-pr996-ui.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KERROKANTASI-UI branch is deployed to platta: https://kerrokantasi-pr996-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://kerrokantasi-pr996-ui.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KERROKANTASI-UI branch is deployed to platta: https://kerrokantasi-pr996-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**KERROKANTASI-UI branch is deployed to platta: https://kerrokantasi-pr996-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://kerrokantasi-pr996-ui.dev.hel.ninja :laughing::tada::tada::tada:

File: src/selectors/oidc.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ * @param {Object} state
+ * @returns User
+ */
+export const getOidcUser = (state) =>
+  state.oidc?.user;
+
+/**
+ *
+ * @param {Object} state
+ * @returns boolean
```
Comment (mikkojamG) [inline src/selectors/oidc.js:12]: I think the return type is incorrect 🤔

File: src/utils/oidcConfig.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+// profiili config
+export const userOidcConfig = {
+  client_id: config.openIdClientId,
+  scope: 'openid profile email',
+  ...commonOidcConfig,
+}
+
+export const adminOidcConfig = {
+  client_id: 'kerrokantasi-ui-staff-dev',
```
Comment (mikkojamG) [inline src/utils/oidcConfig.js:22]: I think should be able to give different client ids for different environment 🤔

File: src/utils/oidcConfig.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  scope: 'openid profile',
+  ...commonOidcConfig,
+}
+
+export const apiTokenClientConfig = {
+  url: 'https://tunnistus.test.hel.ninja/auth/realms/helsinki-tunnistus/protocol/openid-connect/token',
```
Comment (mikkojamG) [inline src/utils/oidcConfig.js:28]: Url should probably come from config

File: src/utils/oidcConfig.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  queryProps: {
+    grantType: 'urn:ietf:params:oauth:grant-type:uma-ticket',
+    permission: '#access',
+  },
+  audiences: ['kerrokantasi-api-dev'],
```
Comment (mikkojamG) [inline src/utils/oidcConfig.js:33]: Audiences should probably also come from config.

**Review by mikkojamG — CHANGES_REQUESTED** [review]
Looking good so far 👌 Left some comments on couple of things I noticed. Also could run format for all files, I think couple of them look unformatted atm 👀

**Comment by terovirtanen** [comment]
**KERROKANTASI-UI branch is deployed to platta: https://kerrokantasi-pr996-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://kerrokantasi-pr996-ui.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KERROKANTASI-UI branch is deployed to platta: https://kerrokantasi-pr996-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**KERROKANTASI-UI branch is deployed to platta: https://kerrokantasi-pr996-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://kerrokantasi-pr996-ui.dev.hel.ninja :laughing::tada::tada::tada:

File: __tests__/Auth/loginCallback.react-test.jsx [hunk 5]
```diff
@@ -1,60 +1,30 @@
...
+  }
 
-  function getWrapper(props) {
+  function renderComponent(props) {
     const defaultProps = {
-      dispatch,
+      history: createBrowserHistory(),
     };
-    return shallow(<UnconnectedCallbackPage {...defaultProps} {...props} />);
+    render(<LoginProvider {...loginProviderProps}><UnconnectedLoginCallback {...defaultProps} {...props} /></LoginProvider>);
```
[the remark this answers is not in the document]
Comment (arpargo) [reply __tests__/Auth/loginCallback.react-test.jsx:21]: Implemented

File: server/render-middleware.js [hunk 6]
```diff
@@ -85,9 +85,11 @@ export default function renderMiddleware(settings) {
     ) {
       return next();
     }
+    /*
```
[the remark this answers is not in the document]
Comment (arpargo) [reply server/render-middleware.js]: Done

File: src/selectors/oidc.js (hunk 1 above)
Comment (arpargo) [reply src/selectors/oidc.js]: Fixed

File: src/utils/oidcConfig.js (hunk 2 above)
Comment (arpargo) [reply src/utils/oidcConfig.js]: This has also been simplified as we decided not to implement two types of login paths.

File: src/utils/oidcConfig.js (hunk 3 above)
Comment (arpargo) [reply src/utils/oidcConfig.js]: Fixed

File: src/utils/oidcConfig.js (hunk 4 above)
Comment (arpargo) [reply src/utils/oidcConfig.js]: fixed.

**Comment by terovirtanen** [comment]
**KERROKANTASI-UI branch is deployed to platta: https://kerrokantasi-pr996-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**KERROKANTASI-UI branch is deployed to platta: https://kerrokantasi-pr996-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://kerrokantasi-pr996-ui.dev.hel.ninja :laughing::tada::tada::tada:

File: __tests__/Auth/loginCallback.react-test.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (mikkojamG) [reply __tests__/Auth/loginCallback.react-test.jsx:21]: Could still use it here in the test? 🤔

File: src/utils/oidcConfig.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  ...commonOidcConfig,
+}
+
+export const apiTokenClientConfig = {
+  url: config.openIdApiTokenUrl,
+  queryProps: {
+    grantType: 'urn:ietf:params:oauth:grant-type:uma-ticket',
+    permission: '#access',
+  },
+  audiences: [config.opein],
```
Comment (mikkojamG) [inline src/utils/oidcConfig.js:27]: Typo here.

**Review by mikkojamG — CHANGES_REQUESTED** [review]

**Comment by terovirtanen** [comment]
**KERROKANTASI-UI branch is deployed to platta: https://kerrokantasi-pr996-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://kerrokantasi-pr996-ui.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KERROKANTASI-UI branch is deployed to platta: https://kerrokantasi-pr996-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Review by mikkojamG — APPROVED** [review]

**Comment by terovirtanen** [comment]
**KERROKANTASI-UI branch is deployed to platta: https://kerrokantasi-pr996-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://kerrokantasi-pr996-ui.dev.hel.ninja :laughing::tada::tada::tada:

**Pull request merged**